common/index.ts

```diff
@@ -145,4 +145,22 @@ export const LAYER_ICON_TYPE_MAP: { [key: string]: string } = {
 
 export enum TOOLTIP_STATE {
   DISPLAY_FEATURES = 'DISPLAY_FEATURES',
+  FILTER_DRAW_SHAPE = 'FILTER_DRAW_SHAPE',
 }
+
+export enum FILTER_DRAW_MODE {
+  NONE = 'none', // draw filter is inactive
+  POLYGON = 'polygon', // Filter is active and set to draw polygon
+}
+
+export interface DrawFilterProperties {
+  relation?: string;
+  mode: FILTER_DRAW_MODE;
+  filterLabel?: string;
+}
```

Comment:
How about introduce a prop `isDrawActive`, the mode only contains different type of draw.

```
enum FILTER_DRAW_MODE {
  POLYGON = 'polygon', // Filter is active and set to draw polygon
  CIRCLE='circle',
}

export interface DrawFilterProperties {
  relation?: string;
  isDrawActive: boolean
  mode: FILTER_DRAW_MODE;
  filterLabel?: string;
}

```

Reply:
Good suggestion. Then we have to maintain two variable for 1 behavior. For ex: if draw is false, then i have to reset mode to null, similarly if we want to enable draw behavior, we have to set to true and set mode accordingly.